File: boats_to_save_people.cpp
class Solution {
public:
    int numRescueBoats(vector<int>& people, int limit) {
        sort(people.begin(), people.end());
        int ans = 0;

        int i = 0, j = people.size()-1;
        while(i <= j) {
            if(people[i] + people[j] <= limit) {
                i++;
            }
            ans++;
            j--;
        }
        return ans;
    }
};

// Time Complexity : O(n*log(n))
// Space Complexity : O(1)

File: boats_to_save_people.go
func numRescueBoats(people []int, limit int) int {
    sort.Ints(people);

    i, j, ans := 0, len(people)-1, 0

    for i <= j {
        if people[i] + people[j] <= limit {
            i++
        }
        j--
        ans++
    }
    return ans
}

// Time Complexity : O(n*log(n))
// Space Complexity : O(1)
